src/pages/Punch/PunchPage.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useEffect, useState } from 'react';
+import { Route, Switch } from 'react-router-dom';
+import Axios from 'axios';
+import EdsIcon from '../../components/icons/EdsIcon';
+import FooterButton from '../../components/navigation/FooterButton';
+import Navbar from '../../components/navigation/Navbar';
+import NavigationFooter from '../../components/navigation/NavigationFooter';
+import withAccessControl from '../../services/withAccessControl';
+import { COLORS } from '../../style/GlobalStyles';
+import removeSubdirectories from '../../utils/removeSubdirectories';
+import useCommonHooks from '../../utils/useCommonHooks';
+import ClearPunch from './ClearPunch/ClearPunch';
+import { PunchItem } from '../../services/apiTypes';
+import { AsyncStatus } from '../../contexts/McAppContext';
+import VerifyPunch from './VerifyPunch/VerifyPunch';
+import SkeletonLoadingPage from '../../components/loading/SkeletonLoader';
+import ErrorPage from '../../components/error/ErrorPage';
+
+const PunchPage = (): JSX.Element => {
+    const { api, params, path, history, url } = useCommonHooks();
+    const [punch, setPunch] = useState<PunchItem>();
+    const [fetchPunchStatus, setFetchPunchStatus] = useState<AsyncStatus>(
+        AsyncStatus.LOADING
+    );
+
+    useEffect(() => {
+        const source = Axios.CancelToken.source();
+        (async (): Promise<void> => {
+            try {
+                const punchFromApi = await api.getPunchItem(
+                    params.plant,
+                    params.punchItemId,
+                    source.token
+                );
+                setPunch(punchFromApi);
+                setFetchPunchStatus(AsyncStatus.SUCCESS);
+            } catch {
+                setFetchPunchStatus(AsyncStatus.ERROR);
+            }
+        })();
+        return (): void => {
+            source.cancel();
+        };
+    }, [api, params]);
+
+    const determineComponentToRender = (): JSX.Element => {
+        if (punch != undefined) {
+            return punch.clearedAt ? <VerifyPunch /> : <ClearPunch />;
```

Comment:
Ettersom vi laster inn punchen her, tenker jeg at vi kan passe inn punchen i de ulike komponentene, sammen med fetchPunchStatus, slik at vi slipper å laste den inn to ganger. 

Da kan vi også fjerne If-en på linje 50, og heller legge inn en asyncpage som viser feilmelding/loading screen på begge disse sidene.